File: docs-web.resources/com.ibm.docs.web.resources/WebContent/js/concord/beans/Participant.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* U.S. Government Users Restricted Rights: Use, duplication or      */
/* disclosure restricted by GSA ADP Schedule Contract with IBM Corp. */
/*                                                                   */
/* ***************************************************************** */

dojo.provide("concord.beans.Participant");
dojo.require("concord.beans.User");

dojo.declare("concord.beans.Participant", null, {
	
	constructor: function(e, clientId,joinTime) {
		this.user = new concord.beans.User(e);
		this.cid = clientId;
		this.joinTime = -1;
		if (joinTime)
			this.joinTime = joinTime;
	},

	getClientId: function() {
		return this.cid;
	},
	
	getUserBean: function() {
		return this.user;
	},
	getJoinTime: function() {
		return this.joinTime;
	}
});

File: docsapp/com.ibm.concord.document.services/src/com/ibm/concord/session/Participant.java
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* IBM Docs Source Materials                                         */
/*                                                                   */
/* (c) Copyright IBM Corporation 2012. All Rights Reserved.          */
/*                                                                   */
/* ***************************************************************** */

package com.ibm.concord.session;

import java.util.Date;

import com.ibm.docs.directory.beans.UserBean;

/**
 * @author gaowwei@cn.ibm.com
 * 
 */
public class Participant
{
  private UserBean user;

  private String clientId;

  private Date reportTime;

  // Add data segment for join time of this participant...
  private Date joinTime;

  private long currentSeq;

  private long clientSeq;

  public Participant(UserBean user, String clientId, Date time, long serverSeq)
  {
    this.user = user;
    this.clientId = clientId;
    this.reportTime = time;
    // Add data segment for join time of this participant...
    this.joinTime = time;
    this.currentSeq = serverSeq;
    this.clientSeq = 0;
  }

  public Date getReportTime()
  {
    return reportTime;
  }

  // Add data segment for join time of this participant...
  public Date getJoinTime()
  {
    return joinTime;
  }

  public void updateReportTime(Date time)
  {
    this.reportTime = time;
  }

  public UserBean getUserBean()
  {
    return user;
  }

  public String getClientId()
  {
    return clientId;
  }

  public long getCurrentSeq()
  {
    return currentSeq;
  }

  public void updateCurrentSeq(long seq)
  {
    currentSeq = seq;
  }

  public long getClientSeq()
  {
    return clientSeq;
  }

  public void updateClientSeq(long seq)
  {
    clientSeq = seq;
  }
}
